modules/apps/object/object-api/src/main/java/com/liferay/object/system/model/listener/SystemObjectDefinitionMetadataModelListener.java

```diff
@@ -179,14 +189,23 @@ private String _getDTOConverterType() {
 	}
 
 	private JSONObject _getPayloadJSONObject(
-			String objectActionTriggerKey, T originalBaseModel, T baseModel,
-			long userId)
+			String objectActionTriggerKey, ObjectDefinition objectDefinition,
+			T originalBaseModel, T baseModel, long userId)
 		throws PortalException {
 
 		String dtoConverterType = _getDTOConverterType();
 
 		return JSONUtil.put(
 			"classPK", baseModel.getPrimaryKeyObj()
+		).put(
+			"extendedProperties",
+			HashMapBuilder.<String, Object>putAll(
+				_objectEntryLocalService.getSystemObjectExtendedProperties(
+					objectDefinition,
+					GetterUtil.getLong(baseModel.getPrimaryKeyObj()))
+			).putAll(
+				EntityExtensionThreadLocal.getExtendedProperties()
```

Comment:
Do you think that `getExtendedProperties` coming after `getSystemObjectExtendedProperties` will be a problem?

Reply:
`getSystemObjectExtendedProperties` will only return values on delete or **update**.. 

`EntityExtensionThreadLocal.getExtendedProperties()` will only return values on add or **update**.

On **update** we need the existing values + updated values, and the `EntityExtensionThreadLocal.getExtendedProperties()` returns the updated values, ​​from the request..

I tested it and it won't be a problem 😄.

Reply:
Got it. Thanks for explaining.